prover/witness_generator/src/scheduler.rs

```diff
@@ -142,14 +133,15 @@ impl JobProcessor for SchedulerWitnessGenerator {
         else {
             return Ok(None);
         };
-        let proof_job_ids = prover_connection
-            .fri_scheduler_dependency_tracker_dal()
-            .get_final_prover_job_ids_for(l1_batch_number)
-            .await;
+        let recursion_tip_job_id = prover_connection
+            .fri_prover_jobs_dal()
+            .get_recursion_tip_proof_job_id(l1_batch_number)
+            .await
+            .unwrap();
```

Comment:
replace unwrap with expect - and mention the l1_batch_number in the error message

Reply:
Regarding that, core DAL [has recently switched](https://github.com/matter-labs/zksync-era/pulls?q=is%3Apr+is%3Aclosed+author%3Aslowli+Wrap+sqlx) to returning `DalResult`s, the idea of which is to capture the called method name and relevant args automatically; thus, it'd be unnecessary to add error context on call sites. Could make sense to convert the prover DAL similarly.